src/main/java/org/commcare/formplayer/services/RestoreFactory.java

```diff
@@ -310,6 +315,15 @@ private UserSqlSandbox restoreUser(boolean skipFixtures) throws SyncRestoreExcep
                         extras
                 );
                 sandbox.writeSyncToken();
+
+                if (!shouldPurge) {
+                    String newSandboxLocations = UserUtils.getUserLocationsByDomain(domain, sandbox);
+                    if (!oldSandboxLocations.isEmpty() && !oldSandboxLocations.equals(newSandboxLocations)) {
```

Comment:
nit: you can directly assign `hasLocationChanged` to this statement and remove if/else

Reply:
I'm not sure what you meant by this - are you suggesting I reset `hasLocationChanged` to false first and then set it to true if the locations have changed?

Reply:
you can do `hasLocationChanged = !oldSandboxLocations.isEmpty() && !oldSandboxLocations.equals(newSandboxLocations)` instead of if/else.